quetz/main.py

```diff
@@ -1595,9 +1591,8 @@ async def serve_path(
         if channel_proxylist and package_name and package_name in channel_proxylist:
             return RedirectResponse(f"{channel.mirror_channel_url}/{path}")
 
-    if is_package_request and pkgstore_support_url:
-        # we have to ignore type checking here right now, sorry
-        return RedirectResponse(pkgstore.url(channel.name, path))  # type: ignore
+    if is_package_request and pkgstore.support_redirect:
```

Comment:
This is cleaner than my previous implementation but it only applies to package files (`.tar.bz2` and `.conda`).

As far as can see, json files (repodata.json) are streamed. There is no redirect. That's what the function does below. Or?

For LocalStore, I think it would make sense to use RedirectResponse as well.

Reply:
Yeah, the reasons for that is that we need to add the headers for cache control, and we try to do some more intelligent stuff around gzip encoding of these JSON files.
The JSON files without GZIP encoding are really large (>100Mb vs ~15Mb).